sway-core/src/error.rs

```diff
@@ -115,6 +115,10 @@ impl<T> From<Result<T, TypeError>> for CompileResult<T> {
 }
 
 impl<T> CompileResult<T> {
+    pub fn is_ok(&self) -> bool {
+        self.value.is_some() && self.warnings.is_empty() && self.errors.is_empty()
```

Comment:
```suggestion
        self.value.is_some() && self.errors.is_empty()
```
We can have a successful compilation result but still get warnings - should the `self.warnings.is_empty()` part of this condition be removed?